pykeepass/kdbx_parsing/common.py

```diff
@@ -188,6 +195,34 @@ def compute_master(context):
     return master_key
 
 
+def populate_custom_data(kdbx, d):
+    if len(d.keys()) > 0:
+        vd = Container(
+            version=b'\x00\x01',
+            dict=d,
+        )
+        kdbx.header.value.dynamic_header.update(
+            {
+                "public_custom_data":
+                    Container(
+                        id='public_custom_data',
+                        data=vd,
+                        next_byte=0xFF,
+                    )
+            }
+        )
+    else:
+        # Removing header entirely
+        if "public_custom_data" in kdbx.header.value.dynamic_header:
+            del kdbx.header.value.dynamic_header["public_custom_data"]
+
+    # Beyond Python 3.7, construct makes the base class of a Container be `dict` instead of `OrderedDict`
```

Comment:
This is very unfortunate: the `construct` library dynamically changes which methods are available on a `Container` based on the version the Python interpreter reports.

If you run Python 3.5, you get all the `OrderedDict` methods. If you run Python 3.11 you get only `dict` methods.